server/athenian/api/align/goals/templates.py

```diff
@@ -1,37 +1,58 @@
+from __future__ import annotations
+
+from typing import TypedDict
+
 from athenian.api.models.web import JIRAMetricID, PullRequestMetricID, ReleaseMetricID
+from athenian.api.models.web.goal_template import GoalTemplateMetricID
 
-TEMPLATES_COLLECTION = {
+TEMPLATES_COLLECTION: dict[int, _TemplateDefinition] = {
```

Comment:
```suggestion
TEMPLATES_COLLECTION: dict[int, TemplateDefinition] = {
```

Let's make it a fully-featured public class.